InterfaceFortran/TasmanianSG.f90

```diff
@@ -685,7 +685,7 @@ function tsgGetComplexHierarchicalCoefficients(gridID) result(c)
     allocate(c_real(2*tsgGetNumOutputs(gridID)*tsgGetNumPoints(gridID)))
     call tsgghc(gridID, c_real)
     do i = 1,size(c)
-      c(i) = complex( c_real(i), c_real(i + size(c)) )
+      c(i) = cmplx( c_real(i), c_real(i + size(c)) )
```

Comment:
https://software.intel.com/en-us/node/679123
`If kind is present, the kind parameter is that specified by kind; otherwise, the kind parameter is that of default real type.`
You need to have:
`c(i) = cmplx( c_real(i), c_real(i + size(c)), 8 )`